src/main/scala/io/iohk/ethereum/ledger/BlockPreparator.scala

```diff
@@ -254,14 +255,26 @@ class BlockPreparator(
 
         validatedStx match {
           case Right(_) =>
-            val TxResult(newWorld, gasUsed, logs, _, _) = executeTransaction(stx, blockHeader, worldForTx)
+            val TxResult(newWorld, gasUsed, logs, rd, vmError) = executeTransaction(stx, blockHeader, worldForTx)
+
+            val (status, returnData) =
+              if (blockchainConfig.ethCompatibilityMode) (None, None)
+              else (
+                Some(vmError match {
+                  case Some(WithReturnCode(returnCode)) => returnCode
+                  case Some(OutOfGas) => ByteString(0x05)
```

Comment:
Can we introduce a few constants with meaningful names, instead of using these magic numbers `0x05` etc?